commit/tokenprice/outcome.go

```diff
@@ -0,0 +1,169 @@
+package tokenprice
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-ccip/shared"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+// getConsensusObservation Combine the list of observations into a single consensus observation
+func (p *processor) getConsensusObservation(
+	aos []shared.AttributedObservation[Observation],
+) (ConsensusObservation, error) {
+	aggObs := aggregateObservations(aos)
+
+	if len(aggObs.FChain) < p.bigF {
+		return ConsensusObservation{},
+			fmt.Errorf("not enough observations for FChain")
+	}
+
+	fMin := make(map[cciptypes.ChainSelector]int)
+	for chain := range aggObs.FChain {
+		fMin[chain] = p.bigF
+	}
+	fChains := shared.GetConsensusMap(p.lggr, "fChain", aggObs.FChain, fMin)
```

Comment:
Lets add comments for the f usage on all of these calls; consensus on the fChain map uses the role DON F value because all nodes can observe the home chain.